### ISSUE TITLE
How to use tf.Print / tf.print to show experience tuples in DQN

### ISSUE THREAD

**Issue by AntonioSerrano** [openai/baselines#1091]
I'm using a DQN from Baselines and Tensorflow 1.13. Right now I'm debugging my code trying to figure out how the input data for the Neural Network (i.e. the experience tuples) look like. In baselines/baselines/build_graph.py, I tried this around line 386:

        obs_t_input = make_obs_ph("obs_t")
        act_t_ph = tf.placeholder(tf.int32, [None], name="action")
        rew_t_ph = tf.placeholder(tf.float32, [None], name="reward")
        obs_tp1_input = make_obs_ph("obs_tp1")
        done_mask_ph = tf.placeholder(tf.float32, [None], name="done")
        importance_weights_ph = tf.placeholder(tf.float32, [None], name="weight")
        
        print_op = tf.Print(obs_t_input.get(), [obs_t_input.get(), act_t_ph, rew_t_ph, obs_tp1_input.get(), done_mask_ph], "Experience tuples: ")

        # q network evaluation
        with tf.control_dependencies([print_op]):
            q_t = q_func(obs_t_input.get(), num_actions, scope="q_func", reuse=True)  # reuse parameters from act
        q_func_vars = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES, scope=tf.get_variable_scope().name + "/q_func")

But it does not work. Nothing is printed. I'm afraid that I'm not being able to include the print operation "print_op" in the graph. I expect to print the tuple (state, action, reward, nex_state, done) multiplied by the batch size at training. Any clue about how to print these data?

BTW, [tf.Print()](https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/Print) is supposed to be deprecated. I tried something similar with [the new tf.print()](https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/print) but I got identical results:

        print_op = tf.print("Tuple: ", obs_t_input, act_t_ph, rew_t_ph, obs_tp1_input, done_mask_ph, output_stream=sys.stdout, sep=',')

        # q network evaluation
        with tf.control_dependencies([print_op]):
            q_t = q_func(obs_t_input.get(), num_actions, scope="q_func", reuse=True)  # reuse parameters from act

Thx.

**Comment by DanielTakeshi** [comment]
I would recommend using `stable-baselines`.

**Comment by AntonioSerrano** [comment]
Thank you, Daniel. Actually, I tried `stable-baselines` before and I personally find it better than `openai baselines`. But unfortunately, I cannot change the latter implementation in the project I'm working on.

After much research, I finally found out how to print variables out of those theano-like functions included in the `baselines/baselines/deepq/build_graph.py`. One should follow these steps to do so:
1. In `baselines/baselines/deepq/build_graph.py`: by the end of the `build_train()` function, in the U.function, replace the line
```{python}
outputs=td_error,
```
by this other line
```{python}
outputs=(td_error, obs_t_input.get(), act_t_ph, rew_t_ph, obs_tp1_input.get(), done_mask_ph, importance_weights_ph),
```

2. In `baselines/baselines/deepq/deepq.py`: by the end of the `learn()` function, replace the line
```{python}
td_errors = train(obses_t, actions, rewards, obses_tp1, dones, weights)
```
by this other line
```{python}
td_errors, obs_t, act_t, rew_t, obs_t1, done_flag, importance_weights  = train(obses_t, actions, rewards, obses_tp1, dones, weights)
```

Then, print these variables as you like e.g.
```{python}
print("obs_t: {}".format(obs_t))
print("act_t: {}".format(act_t))
print("rew_t: {}".format(rew_t))
print("obs_t1: {}".format(obs_t1))
print("done_flag: {}".format(done_flag))
```
